### DISCUSSION TITLE
New error from pyright after updating,  Type "itemList[Unknown]" is already specialized, possible regression?

### DISCUSSION THREAD

**Discussion by Serhiy1** [microsoft/pyright#5619] [Q&A] [answered]
Hi there. I recently Jumped onto a project that I was working on a few months ago and found that all the type hints were broken. 

Python version: 3.8

I made this minimal reproducible that captures the root cause of the breakage. 

```python
from functools import total_ordering
from typing import ( MutableSequence, TypeVar)

T = TypeVar('T')

@total_ordering
class itemList(MutableSequence[T]):
    ...

TS = TypeVar('TS')
class HigherOrderList(itemList[TS]):
    ...


class Foo:
    ...

class FooList(HigherOrderList[Foo]):
    ...
```

This snippet of code gives the following error. 
```
Type "itemList[Unknown]" is already specializedPylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
```

There seem to be two ways to fix this, removing the `@total_ordering` decorator or downgrading the pylance extension version to one that was released two months ago. Both are not ideal.

I have a suspicion that the cause of the issue is `@total_ordering`  does not have any type hints and resolves to `(function) def total_ordering(cls: Unknown) -> Unknown`

**Comment by erictraut** [comment, marked as answer]
Thanks for reporting the issue. Although it's not obvious, this is related to issue https://github.com/microsoft/pyright/issues/5579. A fix is forthcoming in the next release of pyright (1.1.320).
